src/algebra/group_with_zero/basic.lean

```diff
@@ -379,20 +379,40 @@ by rw [← mul_assoc, mul_inverse_cancel x h, one_mul]
 lemma inverse_mul_cancel_left (x y : M₀) (h : is_unit x) : inverse x * (x * y) = y :=
 by rw [← mul_assoc, inverse_mul_cancel x h, one_mul]
 
-lemma mul_inverse_rev {M₀ : Type*} [comm_monoid_with_zero M₀] (a b : M₀) :
-  ring.inverse (a * b) = ring.inverse b * ring.inverse a :=
+variables (M₀)
+
+@[simp] lemma inverse_one : inverse (1 : M₀) = 1 :=
+inverse_unit 1
+
+@[simp] lemma inverse_zero : inverse (0 : M₀) = 0 :=
+by { nontriviality, exact inverse_non_unit _ not_is_unit_zero }
+
+variables {M₀}
+
+lemma mul_inverse_rev' (a b : M₀) (h : commute a b) :
```

Comment:
In many other cases we put lemmas with assumption `commute a b` into `commute` namespace and make `a` and `b` implicit.

Reply:
I can't find any similar lemmas in the `commute` namespace, so I'm inclined to leave it here. I've made `a b` implicit though, as you suggest.